sdk/helper/automatedrotationutil/fields.go

```diff
@@ -104,3 +114,77 @@ func AddAutomatedRotationFields(m map[string]*framework.FieldSchema) {
 		m[name] = schema
 	}
 }
+
+// HandleRotationJobOperation is a helper method wrapper to the two individual helper methods
+// that register and deregister a rotation job each, and returns a standardized error response
+// with a warning in the event of an unexpected error.
+// revertStorageFunc is a function that should revert the original config back in storege before
+// any fields were updated.
+func (p *AutomatedRotationParams) HandleRotationJobOperation(ctx context.Context, b *framework.Backend, req *logical.Request, revertStorageFunc func() error) *logical.Response {
+	if p.DisableAutomatedRotation {
+		// Disable automated rotation and deregister credentials if required.
+		if err := p.HandleDeregisterRotationJob(ctx, b, req); err != nil {
+			resp := logical.ErrorResponse("error deregistering rotation job but config was successfully updated: %s", err)
+			resp.AddWarning("config was successfully updated despite failing to disable automated rotation")
```

Comment:
Does this warning still make sense if we revert the storage below? I'm thinking of a user seeing the warning, but not knowing that the storage was actually reverted

Reply:
This response is only returned if reverting the storage fails too. If reverting the storage succeeds then we only need to return a standard error as the whole call should fail from the plugin side.